broker-core/src/main/java/io/zeebe/broker/clustering/base/topology/TopologyManagerImpl.java

```diff
@@ -145,104 +119,100 @@ public void onStateChange(Raft raft, RaftState raftState) {
         });
   }
 
-  private class ContactPointsChangeListener implements GossipCustomEventListener {
-    @Override
-    public void onEvent(int senderId, DirectBuffer payload) {
-      final DirectBuffer payloadCopy = BufferUtil.cloneBuffer(payload);
-
-      actor.run(
-          () -> {
-            LOG.trace("Received API event from member {}.", senderId);
-
-            final NodeInfo newMember = readNodeInfo(0, payloadCopy);
-            final boolean memberAdded = topology.addMember(newMember);
-            if (memberAdded) {
-              notifyMemberAdded(newMember);
-            }
-          });
-    }
+  @Override
+  public void event(ClusterMembershipEvent clusterMembershipEvent) {
+    actor.call(
+        () -> {
+          final Member eventSource = clusterMembershipEvent.subject();
+          final Member localNode = atomix.getMembershipService().getLocalMember();
+          switch (clusterMembershipEvent.type()) {
+            case METADATA_CHANGED:
+              LOG.debug(
+                  "Member {} receives metadata change of member {}", localNode, eventSource.id());
+              updatePartitionInfo(eventSource);
+              break;
+            case MEMBER_ADDED:
+              LOG.debug("Member {} receives event member {} added", localNode, eventSource.id());
+
+              final Properties newProperties = eventSource.properties();
+              final String replicationAddress = newProperties.getProperty("replicationAddress");
+              final String managementAddress = newProperties.getProperty("managementAddress");
+              final String clientApiAddress = newProperties.getProperty("clientAddress");
+              final String subscriptionAddress = newProperties.getProperty("subscriptionAddress");
+              final ObjectMapper mapper = new ObjectMapper();
```

Comment:
We can reuse the ObjectMapper in this class, simply create one on creation time.